src/simulation/SimulationManager.java

```diff
@@ -385,8 +395,24 @@ public String getCurrentStateStringForProperty(String property) {
         return statePath.getCurNode().stringForProperty(property);
     }
 
+    /**
+     * getCurrentStateDiffString returns the diff between the current and the previous last-committed state.
+     * @return String
+     */
     public String getCurrentStateDiffString() {
```

Comment:
With the added complexity of this function, I think we should add unit tests for it. In particular, to test that it behaves correctly with alt example.

Reply:
I also find it strange that an overloaded method doesn't call another method in the same family. I think these methods should have different names.